indra/newview/llavatarpropertiesprocessor.cpp

```diff
@@ -52,24 +52,23 @@ LLAvatarPropertiesProcessor::~LLAvatarPropertiesProcessor()
 
 void LLAvatarPropertiesProcessor::addObserver(const LLUUID& avatar_id, LLAvatarPropertiesObserver* observer)
 {
+	if (!observer)
+		return;
+
 	// Check if that observer is already in mObservers for that avatar_id
-	observer_multimap_t::iterator it;
+	using pair = std::pair<LLUUID, LLAvatarPropertiesObserver*>;
+	observer_multimap_t::iterator begin = mObservers.begin();
+	observer_multimap_t::iterator end = mObservers.end();
+	observer_multimap_t::iterator it = std::find_if(begin, end, [&](const pair& p)
```

Comment:
might be just `std::find_if(mObservers.begin(), mObservers.end(), [&](const pair& p)`

Reply:
In such case mObservers.end() will be called twice

Reply:
i mean, there is no need in temporary variables "begin" and "end", since they are used only once